### ISSUE TITLE
Training process killed sometimes with ultralytics v8.2.91

### ISSUE THREAD

**Issue by lushikai** [ultralytics/ultralytics#16291]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

![image](https://github.com/user-attachments/assets/bd93af1d-0469-4c6a-a7b1-3c89f2df0acb)

### Environment

ultralytics v8.2.91
python 3.8.19
torch 2.2.1
cuda 12.4
cudnn 9.1
gpu memory 8G
training batch 16

### Minimal Reproducible Example

```python
from ultralytics import YOLO

# Load a model
model = YOLO("ultralytics/cfg/models/v8/xxx.yaml")  # build a new model from YAML

# Train the model
results = model.train(data="ultralytics/cfg/datasets/xxx.yaml",
                      epochs=300,
                      imgsz=640,
                      batch=16,
                      optimizer='SGD',
                      pretrained=False,
                      name='train',
                      )


```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @lushikai, thank you for bringing this to our attention and for using Ultralytics 🚀! This is an automated response to guide you while an Ultralytics engineer takes a closer look.

For 🐛 bug reports, it's great that you've provided a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). This will definitely help us debug the issue. If you have additional context or logs, feel free to update the issue.

In the meantime, you might want to check our [Docs](https://docs.ultralytics.com) for any potential solutions, especially our [Model Training Tips](https://docs.ultralytics.com/guides/model-training-tips/).

Join our Ultralytics community for more support and insights:
- Real-time chat on [Discord](https://ultralytics.com/discord) 🎧
- In-depth discussions on [Discourse](https://community.ultralytics.com)
- Share and learn on our [Subreddit](https://reddit.com/r/ultralytics)

## Install

Ensure that you have installed the `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a compatible [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).

```bash
pip install ultralytics
```

## Environments

YOLOv8 is flexible and can run across various environments:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If the status badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. We continuously test on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for your patience and understanding 🌟!

**Comment by Y-T-G** [comment]
Probably due to running out of system memory (RAM).

**Comment by lushikai** [comment]
@Y-T-G thanks for your quick reply! So how can i deal with?

**Comment by Y-T-G** [comment]
Can you provide the output after running this command in terminal: `yolo checks`?

**Comment by lushikai** [comment]
![image](https://github.com/user-attachments/assets/4a6938b8-f2e9-4d30-a56b-5e556b678ce6)

**Comment by Y-T-G** [comment]
You can try [creating a swap file](https://docs.rackspace.com/docs/create-a-linux-swap-file).

You can make the size of it be 8GB or 16GB.

**Comment by lushikai** [comment]
I have solved! 😺

**Comment by glenn-jocher** [comment]
Great to hear you've resolved it! If you have any more questions, feel free to ask.

**Comment by lushikai** [comment]
@glenn-jocher Thanks for your kindness! By the way, the training log below is correct?
![image](https://github.com/user-attachments/assets/43802c1e-0022-4afa-9889-147932a60c3e)
![image](https://github.com/user-attachments/assets/63f397ef-c455-46b2-8a4f-e2d95e91194c)
Why the begining of the log is 0? and my initial learning rate is 0.01.

**Comment by glenn-jocher** [comment]
The initial learning rate might be adjusted due to a warmup phase. The log starting at 0 is typical for epoch indexing. If you have further questions, feel free to ask!

**Comment by glenn-jocher** [comment]
To display results in the console without saving, set `save=False` in `model.val()`.

**Comment by lushikai** [comment]
Thank you for your time! But when i search for the docs https://docs.ultralytics.com/modes/val/#arguments-for-yolo-model-validation, It seems like that `save=False` is not a optional argument. 🍰

**Comment by glenn-jocher** [comment]
@lushikai you're right; `save=False` isn't an option for `model.val()`. To avoid saving, ensure your validation setup doesn't include save-related parameters.
